src/panels/developer-tools/service/developer-tools-service.ts

```diff
@@ -0,0 +1,281 @@
+import {
+  css,
+  CSSResultArray,
+  html,
+  LitElement,
+  property,
+  query,
+} from "lit-element";
+import { LocalStorage } from "../../../common/decorators/local-storage";
+import "../../../components/buttons/ha-progress-button";
+import "../../../components/entity/ha-entity-picker";
+import "../../../components/ha-card";
+import "../../../components/ha-yaml-editor";
+import "../../../components/ha-service-picker";
+import "../../../styles/polymer-ha-style";
+import { HomeAssistant } from "../../../types";
+import "../../../util/app-localstorage-document";
+import "../../../components/ha-service-control";
+import { computeDomain } from "../../../common/entity/compute_domain";
+import { computeObjectId } from "../../../common/entity/compute_object_id";
+import { haStyle } from "../../../resources/styles";
+import memoizeOne from "memoize-one";
+import "../../../components/ha-expansion-panel";
+import { safeLoad } from "js-yaml";
+import type { HaYamlEditor } from "../../../components/ha-yaml-editor";
+
+class HaPanelDevService extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @LocalStorage("panel-dev-service-state-service-data", true)
+  private _serviceData;
+
+  @LocalStorage("panel-dev-service-state-yaml-mode", true)
+  private _yamlMode = false;
+
+  @query("ha-yaml-editor") private _yamlEditor?: HaYamlEditor;
+
+  protected render() {
+    const { target, fields } = this._fields(
+      this.hass.services,
+      this._serviceData?.service,
+      !this._yamlMode
+    );
+
+    return html`
+      <div class="content">
+        <p>
+          ${this.hass.localize(
+            "ui.panel.developer-tools.tabs.services.description"
+          )}
+        </p>
+
+        ${this._yamlMode
+          ? html`<ha-yaml-editor
+              .defaultValue=${this._serviceData}
+              @value-changed=${this._yamlChanged}
+            ></ha-yaml-editor>`
+          : html`<ha-service-control
+              .hass=${this.hass}
+              .value=${this._serviceData}
+              showAdvanced
+              @value-changed=${this._serviceChanged}
+            ></ha-service-control>`}
+
+        <div class="button-row">
```

Comment:
Should these be visible if no service is selected?

Can we use `display: flex;justify-content: space-between;` on the button-row class to move YAML to the right and out of the main focus?